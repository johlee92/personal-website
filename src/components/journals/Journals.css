.journals {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.journals-container {
    margin: 10px 0;
    width: 80%;
    max-width: 720px;
}