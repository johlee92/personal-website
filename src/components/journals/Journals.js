import React, { Component } from 'react';
import './Journals.css';
import PersonalWebsiteContext from '../../personalWebsiteContext';
import JournalCard from '../journalCard/JournalCard.js';

class Journal extends Component {
    static contextType = PersonalWebsiteContext;

    render() {
        const orderedJournals = this.context.journals.sort((journal1, journal2) => {
            return journal2.id - journal1.id;
        })

        const journalCards = orderedJournals.map((journal) => {
            return <JournalCard {...journal} {...this.props} />
        })

        return (
            <div className="journals">
                <div className="journals-container">
                    <p>I'm just a normal guy who enjoys pondering about life. I wanted to share my thoughts and get your feedback to learn and grow together.</p>
                    {journalCards}
                </div>
            </div>
        )
    }
}

export default Journal;