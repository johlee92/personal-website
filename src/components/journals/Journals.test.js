import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Journals from './Journals';

describe(`Journals component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
                <Journals />
            , div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component as expected', () => {
        const tree = renderer
            .create(
                <Journals />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})
