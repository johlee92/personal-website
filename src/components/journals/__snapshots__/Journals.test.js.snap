// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Journals component renders the component as expected 1`] = `
<div
  className="journals"
>
  <div
    className="journals-container"
  >
    <p>
      I'm just a normal guy who enjoys pondering about life. I wanted to share my thoughts and get your feedback to learn and grow together.
    </p>
  </div>
</div>
`;
